### ISSUE TITLE
build failed with "AttributeError: 'module' object has no attribute 'HTTPSHandler'"

### ISSUE THREAD

**Issue by menghan** [ansible/ansible-container#492]
##### ISSUE TYPE

 - Bug Report

##### container.yml 

```
version: "2"
settings:
  conductor_base: ubuntu:16.04
services:
  cowproxy:
    from: ubuntu:16.04
    roles:
      - tini
registries: {}
```

##### OS / ENVIRONMENT

```
Ansible Container, version 0.9.1rc0
Linux, gbyte2, 4.4.0-62-generic, #83-Ubuntu SMP Wed Jan 18 14:10:15 UTC 2017, x86_64
2.7.12 (default, Jul  1 2016, 15:12:24)
[GCC 5.4.0 20160609] /home/menghan/.virtualenvs/common/bin/python
{
  "ContainersPaused": 0,
  "Labels": null,
  "CgroupDriver": "cgroupfs",
  "ContainersRunning": 1,
  "ContainerdCommit": {
    "Expected": "422e31ce907fd9c3833a38d7b8fdd023e5a76e73",
    "ID": "422e31ce907fd9c3833a38d7b8fdd023e5a76e73"
  },
  "InitBinary": "",
  "NGoroutines": 27,
  "Swarm": {
    "Managers": 0,
    "ControlAvailable": false,
    "NodeID": "",
    "Cluster": {
      "Spec": {
        "Labels": null,
        "TaskDefaults": {},
        "Orchestration": {},
        "EncryptionConfig": {
          "AutoLockManagers": false
        },
        "Raft": {
          "HeartbeatTick": 0,
          "ElectionTick": 0
        },
        "CAConfig": {},
        "Dispatcher": {}
      },
      "Version": {},
      "ID": "",
      "CreatedAt": "0001-01-01T00:00:00Z",
      "UpdatedAt": "0001-01-01T00:00:00Z"
    },
    "Nodes": 0,
    "Error": "",
    "RemoteManagers": null,
    "LocalNodeState": "inactive",
    "NodeAddr": ""
  },
  "LoggingDriver": "json-file",
  "OSType": "linux",
  "HttpProxy": "http://odroid1:8888",
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  },
  "DriverStatus": [
    [
      "Root Dir",
      "/var/lib/docker/aufs"
    ],
    [
      "Backing Filesystem",
      "extfs"
    ],
    [
      "Dirs",
      "66"
    ],
    [
      "Dirperm1 Supported",
      "true"
    ]
  ],
  "OperatingSystem": "Ubuntu 16.04.2 LTS",
  "Containers": 2,
  "HttpsProxy": "http://odroid1:8888",
  "BridgeNfIp6tables": true,
  "MemTotal": 16694431744,
  "SecurityOptions": [
    "name=apparmor",
    "name=seccomp,profile=default"
  ],
  "Driver": "aufs",
  "IndexServerAddress": "https://index.docker.io/v1/",
  "ClusterStore": "",
  "InitCommit": {
    "Expected": "949e6fa",
    "ID": "949e6fa"
  },
  "Isolation": "",
  "SystemStatus": null,
  "OomKillDisable": true,
  "ClusterAdvertise": "",
  "SystemTime": "2017-05-04T00:14:59.84178902+08:00",
  "Name": "gbyte2",
  "CPUSet": true,
  "RegistryConfig": {
    "InsecureRegistryCIDRs": [
      "127.0.0.0/8"
    ],
    "IndexConfigs": {
      "docker.io": {
        "Official": true,
        "Name": "docker.io",
        "Secure": true,
        "Mirrors": []
      }
    },
    "Mirrors": []
  },
  "DefaultRuntime": "runc",
  "ContainersStopped": 1,
  "NCPU": 4,
  "NFd": 21,
  "Architecture": "x86_64",
  "KernelMemory": true,
  "CpuCfsQuota": true,
  "Debug": false,
  "ID": "MO5A:ZBG6:VMPK:4ZR7:GIZM:73Q7:S6PY:TMIT:HRP4:SFEN:QAE5:ZHL7",
  "IPv4Forwarding": true,
  "KernelVersion": "4.4.0-62-generic",
  "BridgeNfIptables": true,
  "NoProxy": "",
  "LiveRestoreEnabled": false,
  "ServerVersion": "17.04.0-ce",
  "CpuCfsPeriod": true,
  "ExperimentalBuild": false,
  "MemoryLimit": true,
  "SwapLimit": false,
  "Plugins": {
    "Volume": [
      "local"
    ],
    "Network": [
      "bridge",
      "host",
      "macvlan",
      "null",
      "overlay"
    ],
    "Authorization": []
  },
  "Images": 35,
  "DockerRootDir": "/var/lib/docker",
  "NEventsListener": 0,
  "CPUShares": true,
  "RuncCommit": {
    "Expected": "9c2d8d184e5da67c95d601382adf14862e4f2228",
    "ID": "9c2d8d184e5da67c95d601382adf14862e4f2228"
  }
}
{
  "KernelVersion": "4.4.0-62-generic",
  "Arch": "amd64",
  "BuildTime": "2017-04-03T18:07:42.793330938+00:00",
  "ApiVersion": "1.28",
  "Version": "17.04.0-ce",
  "MinAPIVersion": "1.12",
  "GitCommit": "4845c56",
  "Os": "linux",
  "GoVersion": "go1.7.5"
}

```
##### SUMMARY

ansible-container --debug build -vvv failed with BuildError, detail error stack is 
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_K5RrUO/ansible_module_get_url.py", line 500, in <module>
    from ansible.module_utils.urls import *
  File "/tmp/ansible_K5RrUO/ansible_modlib.zip/ansible/module_utils/urls.py", line 414, in <module>
AttributeError: 'module' object has no attribute 'HTTPSHandler'
```

##### STEPS TO REPRODUCE

```
git clone https://github.com/menghan/ansible-container-bugreport
pip setup.py develop  # at the latest develop, commit 97513f77f253740650c798f3cc2b5d436458ee25
pip install docker
sudo ansible-container --debug build -vvv
```

##### EXPECTED RESULTS

build success
##### ACTUAL RESULTS

build failure

see all output at https://gist.github.com/menghan/87cbeb9cbebe31eedd71933f53fe372d

**Comment by chouseknecht** [comment]
@menghan 

Thank for using Ansible Container. We appreciate you taking the time to open an issue, and share your feedback. 

I was able to reproduce this. It comes down to the urllib2 module not having an HTTPSHandler class. According to the [urllib2 docs](https://docs.python.org/2/library/urllib2.html), "If the Python installation has SSL support (i.e., if the ssl module can be imported), HTTPSHandler will also be added."

After building your project, if I run a conductor container, and a container from ubuntu:16.04, with /usr from the conductor mounted into the ubuntu:16.04 container, I cannot import ssl. 

Here's the result of connecting to the ubuntu:16.04 container via `docker exec`, starting a python shell, and attempting the import:
 
```
$ python
>>> import ssl
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/_usr/lib/python2.7/ssl.py", line 97, in <module>
    import _ssl             # if we can't import it, let the error propagate
ImportError: libssl.so.1.0.0: cannot open shared object file: No such file or directory
 ```
However, if I do the same to the conductor container, it works fine.

**Comment by chouseknecht** [comment]
The issue seems to be how libssl gets installed. The following is a search for libssl within the conductor instance:

```
root@be54410ceea3:/# cd /usr
root@be54410ceea3:/usr# find . -name libssl.so.1.0.0
root@be54410ceea3:/usr# cd ..
root@be54410ceea3:/# find . -name libssl.so.1.0.0
./lib/x86_64-linux-gnu/libssl.so.1.0.0
```

**Comment by chouseknecht** [comment]
@menghan 

Got this to work by modifying your role. Here's my `tasks/main.yml`:

```yaml
---

- name: set arch name
  set_fact:
    arch_name: arm64
  when: ansible_machine == "aarch64"

- name: set arch name
  set_fact:
    arch_name: amd64
  when: ansible_machine == "x86_64"

- name: Update packages
  raw: apt-get update -y

- name: Install ssl
  raw: apt-get install -y libssl-dev

- name: "download tini-{{ tini_version }}"
  get_url:
    url: "https://github.com/krallin/tini/releases/download/{{ tini_version }}/tini-static-{{ arch_name }}"
    dest: /usr/sbin/tini
    owner: root
    group: root
    mode: 0755
    validate_certs: no
```

Note, I'm using the `raw` action to run `apt-get` because the `apt` module does not work, given that it too requires that ssl be installed.

**Comment by menghan** [comment]
@chouseknecht thanks for your help!

one more question: why don't mount /lib from the conductor into the ubuntu:16.04 container?  Then libssl.so.1.0.0 can be found.

**Comment by ddelbondio** [comment]
@chouseknecht this is pretty similar to the issue fixed in #481 only with a twist.
The ubuntu image actually has a the correct file (`/_usr/lib/x86_64-linux-gnu/libssl.so`), except is a symlink to `/lib/x86_64-linux-gnu/libssl.so.1.0.0`. This file then doesn't exist in the container at all. 

I am thinking that this could cause all kinds of problems in many different ways. And also your solution could possibly pollute the container somewhat. My initial idea is that we walk the /usr/ dir and look at symlinks for which we then mount the target (recursively) 1:1 as files into the container.
Does that sound reasonable for you? Then I can come up a CR which implements this.
